src/Raven.Server/Documents/Indexes/Persistence/Lucene/Documents/JintLuceneDocumentConverter.cs

```diff
@@ -30,15 +40,37 @@ protected override int GetFields<T>(T instance, LazyStringValue key, object docu
                 newFields++;
             }
 
+            IndexField allFields = null;
+            _fields.TryGetValue(Constants.Documents.Indexing.Fields.AllFields, out allFields);
+            var AllFieldsOptions = allFields?.ToIndexFieldOptions();
             foreach ((var property, var propertyDescriptor) in documentToProcess.GetOwnProperties())
             {
+                var actualValue = propertyDescriptor.Value;
+                var actualFieldOption = DefaultFieldOptionds;
+                string actualFieldName = property;
                 if (_fields.TryGetValue(property, out var field) == false)
                 {
-                    field = IndexField.Create(property, DefaultFieldOptionds, DefaultFieldOptionds);
-                    _fields.Add(property, field);
+                    var shouldCreateField = true;
+                    if (actualValue.IsObject())
```

Comment:
This isn't the way to go about doing this. We want to just have this flow naturally from the code.
So when we add the value from an object field, we'll just use that, no need for special behavior at this location